// containers/store/index.js
import { connect } from 'react-redux'
import React, { Component } from 'react'
import R from 'ramda'
import { fetchNews }  from 'actions'
import { Link } from 'react-router'
import Menu from '../menu'

class Store extends Component {
    componentDidMount() {
        this.props.fetchNews(),
            window.scrollTo(0, 0),
            document.title = `Bax cap | News`
    }
    
    renderNews(news,index){
        const shortDescription = `${R.take(60, news.description)}...`
            const image = news.image
        return(
        <a className="card" href="#" key={index}>
                <span className="card-header" style={{ backgroundImage: `url(${image})`}}>
                <span className="card-title">
                    <h3>{news.name}</h3>
                </span>
            </span>
                <span className="card-summary"><Link to={news.link}>{shortDescription}</Link></span>
            <span className="card-meta">Published: feb 20-1th, 2018</span>
        </a>
        )
    }

    render() {
       const {news} = this.props
        return (
            <div>
                <div className="balance-header">
                    <div className="header-overlay">
                       <Menu />
                        <div className="price-header-content">
                            <h2 className="price-header-title">News</h2>
                            <p className="price-header-text"> Последнии новости в Bax-capital</p>
                        </div>{/* header-content */}
                    </div>{/* header-overlay */}
                </div>{/* personal-header */}
                <div className="container1">
                    <div className="cards">
                        {news.map((news, index) => this.renderNews(news, index))}
                </div>
                </div>
                <div className="footer">
                    <div className="wrapper">
                        <div className="footer-container">
                            <div className="footer-menu">
                                <span className="footer-menu-title">Help &amp; Support</span>
                                <button className="footer-menu-btn">read more</button>
                            </div>{/* footer-menu */}
                            <div className="footer-menu">
                                <span className="footer-menu-title">FAQ</span>
                                <button className="footer-menu-btn">read more</button>
                            </div>{/* index-footer-menu */}
                        </div>{/* index-footer-container */}
                        <p className="footer-text">Packages for mobile, IoT, front end, back end, robotics… everything you need to start building amazing things.</p>
                    </div>{/* wrapper */}
                    <p className="footer-copyright">© 2012 – 2018, www.sitename.by</p>
                </div>{/* index-footer */}
            </div>
        )
    }
}
export const getNewsById = (state, id) => R.prop(id, state.news)

export const getNews = state => {
    const news = R.map(id => getNewsById(state, id), state.newsPage.ids)
    return news 
}
    
const mapStateToProps = state => ({
    news: getNews(state)
   
})
const mapDispatchToProps = {
    fetchNews
}
export default connect(mapStateToProps, mapDispatchToProps)(Store)
