// containers/signIn/component.js
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { auth }  from 'actions'
import Request from '../request'

export class FluidInput extends Component {
    constructor(props) {
        super(props);
        this.state = {
            focused: false,
            value: ""
        };
    }
    focusField() {
        const { focused } = this.state;
        this.setState({
            focused: !focused
        });
    }
    handleChange(event) {
        const { target } = event;
        const { value } = target;
        this.setState({
            value: value
        });

        this.props.onInputChange(event.target.name ,event.target.value )
    }
    render() {
        const { type, label, style, id, valid } = this.props;
        const { focused, value } = this.state;

        let inputClass = "fluid-input";
        if (focused) {
            inputClass += " fluid-input--focus";
        } else if (value != "") {
            inputClass += " fluid-input--open";
        }

        return (
            <div className={inputClass} style={style}>
                <div className="fluid-input-holder">
                    <input
                        className="fluid-input-input"
                        type={type}
                        name={id}
                        id={id}
                        onFocus={this.focusField.bind(this)}
                        onBlur={this.focusField.bind(this)}
                        onChange={this.handleChange.bind(this)}
                        autoComplete="off"
                    />
                    <label className="fluid-input-label" forHtml={id}>{label}</label>

                </div>
            </div>
        );
    }
}

FluidInput.defaultProps = {
    valid : true
}

export class Button extends Component {
    render() {
        return (
            <div className={`button ${this.props.buttonClass}`} onClick={this.props.onClick}>
                {this.props.buttonText}
            </div>
        );
    }
}

class SignInContainer extends Component {

    state = {
        user : {},

    }
    onInputChange = (name, value) => {
        const user = {...this.state.user, [name] : value}
        this.setState({user})
    }

    onSubmit = () => {


        const userAuth = Request('GET', this.state.user)
        if(userAuth){
            this.props.auth(userAuth)

        }

    }

    render() {

        const style = {
            margin: "15px 0"
        };
        return (
            <div className="login-container">
                <div className="title">
                    Sign In
                </div>
                <FluidInput type="text" label="name" id="name" style={style} onInputChange={this.onInputChange}/>
                <FluidInput type="password" label="password" id="password" style={style} onInputChange={this.onInputChange}/>
                <Button buttonText="Sign-In" buttonClass="login-button" onClick={this.onSubmit}/>
            </div>
        );
    }
}
const mapDispatchToProps = {
    auth
}


export default connect(null, mapDispatchToProps)(SignInContainer)
