// containers/balance/index.js
import { connect } from 'react-redux'
import React, { Component } from 'react'
import R from 'ramda'
import getCoin from 'selectors'
import { Link } from 'react-router'
import Menu from '../menu'
import SimpleModal from '../modal'
import { Message } from 'semantic-ui-react'
import Mi from '../modal'
import Mi_eth from '../modal_2'
import Mi_coin from '../modal_3'
import { signout } from 'actions'
import { ToastContainer, toast } from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css'

import { fetchPrice } from 'actions'
import { getCryptoCurrency } from 'selectors'

class Balance extends Component {

    componentDidMount() {
        window.scrollTo(0, 0),
            this.props.fetchPrice(),
            document.title = "Bax cap | Profile"
    }

    renderPrice_table_Charts_1(prices, index) {

        let
            percent = prices.percent_change_24h,
            color = percent < 0 ? 'red' : 'rgb(2, 214, 19)'
        return (
            <tr key={index} >
                <td >{prices.symbol}</td>
                <td>0.0000000</td><td><a>0.0000000</a></td>
                <Mi_coin />
            </tr>
        )
    }


    popol = () => toast.warn(`Пополнение валюты USD - временно отсутствует! Скоро мы все исправим`, {
        position: toast.POSITION.BOTTOM_RIGHT
    })
    vivod = () => toast.info(`Ваш баланс равен нулю!`, {
        position: toast.POSITION.BOTTOM_RIGHT
    })
    error = () => toast.error(`Ошибка : баланс равен нулю`, {
        position: toast.POSITION.BOTTOM_RIGHT
    })
    render() {
        const { prices } = this.props
        return (
             !this.props.hideButton ? <div><Menu /><br /> <div className="wrapper"><h1><p style={{ color:'rgba(134, 134, 134, 0.918)'}}>404 - Окончание сесии , авторизируйтесь заново</p></h1> </div></div>:
            <div>
                <div>
                    <div className="personal-header">
                        <div className="header-overlay">
                            <Menu />

                        </div>{/* header-overlay */}
                    </div>{/* personal-header */}
                    <Message visible>
                        Добро пожаловать в личный кабинет | Тех.поддержка - +129324889201 , telegram: @baax-tex
                    </Message>
                    <div className="personal-container">
                        <div className="wrapper">

                            <span className="diagram-tittle">Balance </span><br />

                            <div className="wallet-block">

                                <div className="wallet-et">
                                    <span className="wallet-et-header">ETH balance</span>
                                    <b>Balance:</b>
                                    <span className="wallet-et-sum">0 eth</span>
                                    <b>Your adress:</b>
                                    <span className="wallet-et-adr"><p className='et-to'>0x0074709077B8AE5a245E4ED161C971Dc4c3C8E2B</p></span>
                                    <div className="wallet-rub-row">
                                        <Mi_eth />
                                    </div>{/* wallet-rub-column */}
                                </div>{/* wallet-et */}
                                <div className="wallet-rub">
                                    <span className="wallet-rub-header">RUR &amp; USD</span>
                                    <b>Balance:</b>
                                    <div className="wallet-rub-sum-container">
                                        <span>RUR: 0 р</span>
                                        <span className="wallet-rub-line" />
                                        <span>USD: 0 $</span>
                                    </div>{/* wallet-rub-sum-container */}
                                    <b className="wallet-rub-title">Операции с балансом</b>
                                    <div className="wallet-rub-row">

                                        <Mi />

                                        <div className="wallet-rub-column">
                                            <span>USD</span>
                                            <button className="wallet-rub-btn green-btn" onClick={this.popol}>Пополнить <img src="/img/paypal.svg" width={25} alt="fund" /></button>
                                            <button className="wallet-rub-btn green-btn">Вывести <img src="/img/paypal.svg" width={25} alt="fund" /></button>
                                            <ToastContainer />


                                        </div>{/* wallet-rub-column */}
                                    </div>{/* wallet-rub-row */}
                                </div>{/* wallet-rub */}
                            </div>{/* wallet-block */}
                        </div>{/* wrapper */}

                    </div>{/* personal-container */}
                    <div className="balance-info">
                        <div className="wrapper">

                            <div className="diagram-container">
                                <div className="diagram-left">
                                    <span className="diagram-tittle">Cryptotransfer</span>

                                    <div className="diagram-input-container">
                                        <input type="text" placeholder={1} />
                                        <select>
                                            <option value="rub">Russian Ruble</option>
                                            <option value="usd">USD</option>
                                            <option value="utb">Ethereum</option>
                                        </select>
                                    </div>{/* diagram-input-container */}
                                    <br />
                                    <div className="diagram-input-container">
                                        <button className="wallet-rub-btn orange-btn"  >
                                            <img src="/img/sort.svg" width={20} alt="fund" /> </button>
                                    </div>
                                    {/* diagram-input-container */}
                                    <div className="diagram-input-container">
                                        <input type="text" placeholder="0.00" />
                                        <select>
                                            <option value="rub">Ethereum</option>
                                            <option value="usd">Bitcoin</option>
                                            <option value="rub">Russian Ruble</option>
                                            <option value="usd">USD</option>
                                        </select>
                                    </div>{/* diagram-input-container */}
                                </div>{/* diagram-left */}

                            </div>{/* diagram-container */}
                            <br />
                            <h4><p className='colpb'>Coin balance </p></h4>
                            <hr className="style18" />
                            <div className="balance-tab-header">
                                <table border={1}>
                                    <tbody><tr><td>Name</td><td>Amont</td><td>Orders</td><td>Вывод</td></tr>
                                    </tbody></table>
                            </div>{/* balance-tab */}
                            <div className="balance-tab">
                                <table border={1}>
                                    <tbody>
                                        {prices.map((prices, index) => this.renderPrice_table_Charts_1(prices, index))}
                                    </tbody></table>
                            </div>{/* balance-tab */}
                            <h4><p className='colpb'>Profit for fund</p></h4>
                            <hr className="style18" />
                            <div className="balance-tab-header">
                                <table border={1}>
                                    <tbody><tr><td>Amont</td><td>Balance</td><td>Date</td><td>Profit:</td></tr>
                                    </tbody></table>
                            </div>{/* balance-tab */}
                            <div className="balance-tab-history">
                                <table border={1}>
                                    <tbody>
                                        <tr><td>-</td>
                                            <td>-</td><td><a>-</a></td>
                                            <td>null</td>
                                        </tr>
                                    </tbody></table>
                            </div>{/* balance-tab */}

                        </div>{/* wrapper */}
                    </div>{/* balance-info */}
                    <div className="footer">
                        <div className="wrapper">
                            <div className="footer-container">
                                <div className="footer-menu">
                                    <span className="footer-menu-title">Help &amp; Support</span>
                                    <button className="footer-menu-btn">read more</button>
                                </div>{/* footer-menu */}
                                <div className="footer-menu">
                                    <span className="footer-menu-title">FAQ</span>
                                    <button className="footer-menu-btn">read more</button>
                                </div>{/* footer-menu */}
                            </div>{/* footer-container */}
                            <p className="footer-text">Packages for mobile, IoT, front end, back end, robotics… everything you need to start building amazing things.</p>
                        </div>{/* wrapper */}
                        <p className="footer-copyright">© 2012 – 2018, www.sitename.by</p>
                    </div>{/* footer */}
                </div>

            </div>
        )
    }
}



//Раскомментировать эту константу, когда уберется перезагрузка страниц
const mapStateToProps = state => ({
    hideButton: state.auth.hideButton,
    prices: getCryptoCurrency(state)
})
///

const mapDispatchToProps = {
    signout,
    fetchPrice
}


export default connect(mapStateToProps, mapDispatchToProps)(Balance)
